app/services/state_file/ty23_archiver_service.rb

```diff
@@ -0,0 +1,74 @@
+module StateFile
+  class Ty23ArchiverService
+    INTAKE_MAP = {
+      'az' => StateFileAzIntake,
+      'ny' => StateFileNyIntake,
+    }.freeze
+
+    attr_reader :state_code, :batch_size, :data_source, :tax_year, :current_batch, :cutoff
+
+    def initialize(state_code:, batch_size: 100, cutoff: '2024-06-01')
+      @state_code = state_code
+      @batch_size = batch_size
+      @data_source = INTAKE_MAP[state_code]
+      @tax_year = 2023
+      @cutoff = cutoff
+      @current_batch = nil
+      raise ArgumentError, "#{state_code} isn't an archivable state. Expected one of #{INTAKE_MAP.keys.join(', ')}" unless data_source
+    end
+
+    def find_archiveables
+      @current_batch = ActiveRecord::Base.connection.exec_query(query_archiveable)
+      Rails.logger.info("Found #{current_batch.count} #{data_source} intakes to archive.")
+    end
+
+    def archive_batch
+      archived_ids = []
+      current_batch.each do |record|
+        archive = StateFileArchivedIntake.new(record.without('source_intake_id'))
+        archive.submission_pdf.attach(data_source.find(record['source_intake_id']).submission_pdf.blob)
+        archive.save!
+        archived_ids << record['source_intake_id']
```

Comment:
🤔 i am assuming that we don't want to archive and delete in the same rake task.

maybe it would be helpful to have a method that after the `archive_batch` is run, we verify that the archivable intakes now have an archivable record. 

i don't if that makes sense or is easy.

Reply:
also how were you thinking of saving the intake IDs?

Reply:
not deleting in this task. collecting the ids for logging purposes only. we'll do the actual table truncation in a different effort.